pkg/tun/tun_linux.go

```diff
@@ -0,0 +1,146 @@
+// tun_linux.go: Open a Tunnel (L3 virtual interface) using the Universal TUN/TAP device driver.
+package tun
```

Comment:
Without a blank line between the comment and the `package` line, this will show up as package-level documentation.  Which is mostly-fine, except then it probably shouldn't say `tun_linux.go: `.